packages/vite/src/node/plugins/preAlias.ts

```diff
@@ -14,7 +14,8 @@ export function preAliasPlugin(): Plugin {
       server = _server
     },
     resolveId(id, importer, options) {
-      if (!options?.ssr && bareImportRE.test(id)) {
+      const hasPreBundled = server._optimizeDepsMetadata?.optimized[id]
+      if (!options?.ssr && (bareImportRE.test(id) || hasPreBundled)) {
```

Comment:
**suggestion (micro-perf):** We could swap the or, so the regex will only executed on non short circuit

```suggestion
      if (!options?.ssr && (hasPreBundled || bareImportRE.test(id))) {
```

Reply:
Assuming we apply the suggestion in [this comment](https://github.com/vitejs/vite/pull/5495#discussion_r740002314), I think having the regex first may be better as that's the more common case :thinking:

Reply:
> Assuming we apply the suggestion in [this comment](https://github.com/vitejs/vite/pull/5495#discussion_r740002314), I think having the regex first may be better as that's the more common case 🤔

I agree with you since bare import is more common base.

Reply:
> Assuming we apply the suggestion in [this comment](https://github.com/vitejs/vite/pull/5495#discussion_r740002314), I think having the regex first may be better as that's the more common case 🤔

I agree with you since bare import is more common case.

Reply:
But `hasPreBundled` is calculated before anyways in one line above?!
Also `hasPreBundled` is a `O(1)` access and `bareImportRE.test(id)` is a bit more, cause the regex needs to be executed. You know what I mean?

In both solutions, the logic isn't changed, but in my suggestion the performance is at least 1 nanosecond faster 🤣

Reply:
Nonono, as you can see, hasPreBundled expression has been inside.

Reply:
here
```     
 if (
        !options?.ssr &&
        (bareImportRE.test(id) || server._optimizeDepsMetadata?.optimized[id])
      )
```

Reply:
here
```js
  if (
    !options?.ssr &&
    (bareImportRE.test(id) || server._optimizeDepsMetadata?.optimized[id])
  )
```

Reply:
But do I understand something wrong? Or is `server._optimizeDepsMetadata?.optimized[id]` always faster as `bareImportRE.test(id)`?

Reply:
Yes, i  know what you mean.The former is indeed faster than the latter.But bare import is almost common case, if run `server._optimizeDepsMetadata?.optimized[id]` first, it will have unnecessary performance consumption in most cases.